src/library/scala/sys/process/package.scala

```diff
@@ -157,7 +157,7 @@ package scala.sys {
     *     while(input.read() != -1) count += 1
     *     input.close()
     *   }
-    *   cat ! new ProcessIO(_.close(), byteCounter, _.close())
+    *   cat run new ProcessIO(_.close(), byteCounter, _.close())
```

Comment:
Just using `run` doesn't work in the example because `run` returns immediately, it doesn't block until the process is finished. So the example method will most likely return `0`.

Here's how to block:
def gzFileSize(name: String) = {
  val cat = Seq("zcat", name)
  var count = 0
  def byteCounter(input: java.io.InputStream) = {
    while(input.read() != -1) count += 1
    input.close()
  }
  val p = cat run new ProcessIO(_.close(), byteCounter, _.close())
  p.exitValue() // block until p finished
  count
}